### PR TITLE
feat(tax): Update tax definitions

### PR DESCRIPTION
**Pull request by vincent-pochet**

### CODE REVIEW

File: openapi.yaml [hunk 1]
```diff
@@ -1778,18 +1778,18 @@ paths:
       tags:
         - taxes
       summary: Create a tax
-      description: Create a tax
+      description: This endpoint creates a new tax representing a customizable tax rate applicable to either the organization or a specific customer.
```
Comment (rsempe) [inline openapi.yaml:1781]: Do we already put that it can be defined on the plan or a specific charge?

File: openapi.yaml [hunk 2]
```diff
@@ -1803,60 +1803,70 @@ paths:
     get:
       tags:
         - taxes
-      summary: Find taxes
-      description: Find all taxes in certain organization
+      summary: List all taxes
+      description: This endpoint retrieves all existing taxes representing a customizable tax rate applicable to either the organization or a specific customer.
```
Comment (rsempe) [inline openapi.yaml:1807]: same here

File: openapi.yaml [hunk 3]
```diff
@@ -1803,60 +1803,70 @@ paths:
...
                 $ref: '#/components/schemas/TaxesPaginated'
+              example:
+                taxes:
+                  - lago_id: 41190609-391a-4368-9549-d4348a85a8cd
+                    name: TVA
+                    code: french_standard_vat
+                    description: French standard VAT
+                    rate: '20.0'
+                    applied_to_organization: true
+                    customers_count: 0
```
Comment (rsempe) [inline openapi.yaml:1827]: It makes me think that we should probably add this for `plans_count` and `charges_count`..

File: openapi.yaml (hunk 1 above)
Comment (vincent-pochet) [reply openapi.yaml:1781]: I guess we will update it later

File: openapi.yaml [hunk 4]
```diff
@@ -1890,11 +1900,11 @@ paths:
       tags:
         - taxes
       summary: Delete a tax
-      description: Delete a tax
+      description: This endpoint is used to delete a tax.
```
Comment (rsempe) [inline openapi.yaml:1903]: Maybe we can add something related to the impact for draft and future invoices?

File: openapi.yaml [hunk 5]
```diff
@@ -2663,29 +2673,30 @@ components:
...
           format: '^[0-9]+.?[0-9]*$'
-          description: The rate of the tax.
-          example: '25.0'
+          description: The percentage rate of the tax
+          example: '20.0'
         tax_description:
           type: string
-          description: The description of the tax.
-          example: description
+          description: Internal description of the taxe
```
Comment (rsempe) [inline openapi.yaml:2689]:
```suggestion
          description: Internal description of the tax
```

Comment (rsempe) [reply openapi.yaml:2689]: What does `internal` mean? 😅

File: openapi.yaml [hunk 6]
```diff
@@ -5960,80 +6013,60 @@ components:
...
           type: string
-          description: Unique code used to identify the tax.
-          example: vat_fr
+          description: Unique code used to identify the tax associated with the API request.
+          example: french_standard_vat
         description:
           type: string
-          description: Description of the tax.
-          example: description
+          description: Internal description of the taxe
```
Comment (rsempe) [inline openapi.yaml:6042]:
```suggestion
          description: Internal description of the tax
```

File: src/schemas/TaxBaseInput.yaml [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+    example: 'french_standard_vat'
+  rate:
+    type: string
+    format: '^[0-9]+.?[0-9]*$'
+    description: The percentage rate of the tax
+    example: '20.0'
+  description:
+    type: string
+    nullable: true
+    description: Internal description of the taxe
```
Comment (rsempe) [inline src/schemas/TaxBaseInput.yaml:19]:
```suggestion
    description: Internal description of the tax
```

File: src/schemas/TaxObject.yaml [hunk 8]
```diff
@@ -3,42 +3,43 @@ required:
...
     type: string
-    description: Unique code used to identify the tax.
-    example: 'vat_fr'
+    description: Unique code used to identify the tax associated with the API request.
+    example: 'french_standard_vat'
   description:
     type: string
-    description: Description of the tax.
-    example: 'description'
+    description: Internal description of the taxe
```
Comment (rsempe) [inline src/schemas/TaxObject.yaml:26]:
```suggestion
    description: Internal description of the tax
```

**Review by rsempe — APPROVED** [review]

**Pull request merged**